File: js_implement/sort/bubbleSort.ts
export const bubbleSort = <T>(arr: T[], n: number) => {
    let temp: T
    for (let i = 0; i < n; i++) {
        let flag = false
        for (let j = 0; j < n - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                [arr[j + 1], arr[j]] = [arr[j], arr[j + 1]]
                flag = true
            }
        }
        if (flag) return arr
    }
    return arr
}

File: rust_implement/src/sort/bubble_sort.rs
pub fn bubble_sort(arr: &mut Vec<i32>) {
    let n = arr.len();
    for i in 1..n {
        let mut swapped = false;
        for j in 0..n - i {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
                swapped = true;
            }
        }
        if !swapped {
            break;
        }
    }
}
